### PR TITLE
Added moveHistory in gameObject, also added rooms collection to store round level information

### PR DESCRIPTION
**Pull request by davidgladson**

### CODE REVIEW

File: src/config.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 const local = {
-    mongoURI: 'mongodb://127.0.0.1:27017,127.0.0.1:27018,127.0.0.1:27019/poker-db?' +
-        'replicaSet=rs',
+    mongoURI: 'mongodb://megapokerbot:mega123@cluster0-shard-00-00.vpwpq.mongodb.net:27017,cluster0-shard-00-01.vpwpq.mongodb.net:27017,cluster0-shard-00-02.vpwpq.mongodb.net:27017/poker-db?' +
+        'replicaSet=atlas-hugr40-shard-0&ssl=true&authSource=admin',
```
Comment (avikalpg) [inline src/config.js]: This is actually the production database URI. It is mentioned in the mongoURI environment variable in the YAML files. I don't think it is a good idea to have the development DB the same as the production DB.

File: src/service/gameService.js [hunk 2]
```diff
@@ -1,6 +1,8 @@
 const { PREFLOP, FLOP, TURN, RIVER, phases, DECK } = require('../constants')
 const { distributeChipsToWinners, getHandRanks } = require('./winnerService')
 const { encryptCard, decryptHand } = require('./encryptionService')
+const { Room } = require('../models/Room')
+const func = require('joi/lib/types/func')
```
Comment (avikalpg) [inline src/service/gameService.js]: Is this import used anywhere?

File: src/service/gameService.js [hunk 3]
```diff
@@ -318,7 +320,43 @@ const startNextRound = game => {
...
+        "deck": game.deck,
+        "communityCards": game.communityCards,
+        "allInHands": game.allInHands,
+        "sidePots": game.sidePots,
+        "winners": game.winners,
+        "endedByFold": game.endedByFold,
+        "numBots": game.numBots,
+        "moveHistory": game.moveHistory,
+    })
+    console.log("room saved")
```
Comment (avikalpg) [inline src/service/gameService.js]: Check indentation. You would like to run `npm test` from the root folder to make sure that your code meets the linting requirements of the project.

File: src/models/Room.js [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+        required: true,
+        default: 0
+    },
+    moveHistory: {
+        type: Array,
+        default: []
+    }
+})
+
+const Room = mongoose.model('Room', roomSchema)
```
Comment (avikalpg) [inline src/models/Room.js]: "Room" is not a descriptive name here (this code is not written in the context of the GetMega platform terms). 
Wouldn't "HandHistory" or "GameHistory" be a better name?

File: src/models/Room.js [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+    },
+    numBots: {
+        type: Number,
+        required: true,
+        default: 0
+    },
+    moveHistory: {
+        type: Array,
+        default: []
+    }
```
Comment (avikalpg) [inline src/models/Room.js]: Do you actually need to write all this? 
```suggestion
    ...gameSchema
```

Assuming gameSchema can be extracted from models/Game somehow. Or maybe you can export the gameSchema as well from the Game file.

**Review by avikalpg — CHANGES_REQUESTED** [review]

File: src/config.js (hunk 1 above)
Comment (davidgladson) [reply src/config.js]: reverted back to the local URI

File: src/service/gameService.js (hunk 2 above)
Comment (davidgladson) [reply src/service/gameService.js]: line 4 - yes while writing data to db, func isn't used so removed it

File: src/service/gameService.js (hunk 3 above)
Comment (davidgladson) [reply src/service/gameService.js]: done, used "Format document"

File: src/models/Room.js (hunk 4 above)
Comment (davidgladson) [reply src/models/Room.js]: right yes, renamed it to HandHistory

File: src/models/Room.js (hunk 5 above)
Comment (davidgladson) [reply src/models/Room.js]: got it, removed all of this and imported gameSchema and used it

**Review by avikalpg — APPROVED** [review]

**Pull request merged**